frank-doc-frontend/src/app/views/details/details-search/details-search.component.ts

```diff
@@ -22,46 +22,74 @@ export class DetailsSearchComponent implements OnChanges {
   @Input({ required: true }) frankDocElements!: FrankDoc['elements'] | null;
 
   protected searchQuery = '';
+  protected relatedSearchQuery = '';
   protected filteredElements: FuseResult<Element>[] = [];
+  protected relatedElements: FuseResult<Element>[] = [];
   protected showRelated: boolean = environment.relatedSearchResults;
 
   private appService: AppService = inject(AppService);
   protected getFirstFilter = this.appService.getFirstFilter;
   protected getFirstLabel = this.appService.getFirstLabel;
 
   private elementsList: Element[] = [];
+  private exclusiveElementsList: Element[] = [];
   private fuse: Fuse<Element> = new Fuse(this.elementsList, fuseOptions);
+  private fuseRelated: Fuse<Element> = new Fuse(this.elementsList, { ...fuseOptions, shouldSort: false });
 
   ngOnChanges(changes: SimpleChanges): void {
     if (changes['frankDocElements']) {
       this.elementsList = Object.values(this.frankDocElements ?? {}).filter((element) => !element.deprecated);
       this.fuse.setCollection(this.elementsList);
     }
     if (changes['element']) {
-      if (this.searchQuery === '') {
-        const firstElementNamePart = this.element?.name.split(/(?=[A-Z])/)[0];
-        this.searchQuery = firstElementNamePart ?? '';
+      if (this.element) {
+        const firstElementNamePart = this.element.name.split(/(?=[A-Z])/)[0];
```

Comment:
Maybe also split on numbers? We have a lot of pipes with a '2' in them.
Might be nice to create a variable (or method) for this regex for readability.